src/main/jbake/content/blog/2017/07/jaeger-with-security-proxy.adoc

```diff
@@ -0,0 +1,252 @@
+Once the Keycloak server is up and running, let's create a realm for Jaeger:
+
+* Login into Keycloak (http://<YOUR_IP>:8080/auth/admin/master/console) with `admin` as username and `password` as password
+* On the top left corner, mouse over the `Select realm` box and click `Add realm`. Name it `jaeger` and click `Create`
+* On `Clients`, click `Create` and set `proxy-jaeger` as the name and save it
+* Set the `Access Type` to `confidential` and `*` as `Valid Redirect URIs` and save it. You might want to fine tune this in a 
+production environment, otherwise you might be open to an attack known as 
+link:https://www.owasp.org/index.php/Unvalidated_Redirects_and_Forwards_Cheat_Sheet["Unvalidated Redirects and Forwards"].
+* Open the `Installation` tab and select `Keycloak OIDC JSON` and copy the JSON that is shown. It should look like this:
+
+```json
+{
+  "realm": "jaeger",
+  "auth-server-url": "http://192.168.2.111:8080/auth",
+  "ssl-required": "external",
+  "resource": "proxy-jaeger",
+  "credentials": {
+    "secret": "7f201319-1dfd-43cc-9838-057dac439046"
+  }
+}
+```
+
+And finally, let's create a role and a user, so that we can login to Jaeger's Query service:
+
+* Under the `Configure` left-side menu, open the `Roles` page and click `Add role`
+* As role name, set `user` and click `Save`
+* Under the `Manage` left-side menu, open the `Users` page and click `Add user`
+* Fill out the form as you wish and set `Email verified` to `ON` and click on `Save`
+* Open the `Credentials` tab for this user and set a password (temporary or not).
+* Open the `Role mappings` tab for this user, select the role `user` from the `Available Roles` list and click `Add selected`
+
+=== Preparing OpenShift
+
+For this demo, we assume you have an OpenShift cluster running already. If you don't have, you might want to checkout tools
+like link:https://github.com/minishift/minishift[`minishift`]. If you are running a recent version of Fedora, CentOS or
+Red Hat Enterprise Linux you might want to install the package `origin-clients` and run `oc cluster up --version=latest`.
+This should get you a basic OpenShift cluster running locally.
+
+To make it easier for our demonstration, we'll add `cluster-admin` rights to our `developer` user and we'll create 
+the Jaeger namespace:
+
+```bash
+oc login -u system:admin
+oc new-project jaeger
+oc adm policy add-cluster-role-to-user cluster-admin developer
+oc login -u developer
+```
+
+=== Preparing the Jaeger OpenShift template
+
+We'll use the link:https://github.com/jaegertracing/jaeger-openshift/blob/master/production/jaeger-production-template.yml[Jaeger OpenShift Production template]
+as the starting point: either clone the entire repository, or just get a local version of the template.
+
+The first step is to add the sidecar container to the `query-deployment` object. Under the `containers` list, after we
+specify the `jaeger-query`, let's add the sidecar:
+
+```yml
+        - image: jpkroehling/keycloak-proxy
```

Comment:
If that won't get merged, I'll probably add a new section to this blog post, with instructions on how to build that image.